File: GO/graph/kruskal.go
package graph

import(
	"encoding/json"
	"sort"
	"log"
)

func ApplyKruskal(g Graph) *Graph {
	new_Graph := new(Graph)
	indexes := g.GetIndexes()
	size := len(indexes)
	conections := make([]int,size)
	for i := 0; i < size; i++ {
		conections[i] = -1
	}
	sort.Sort(g)
	queue_sorted := g.GetEdges()
	log.Println(indexes)
	log.Printf("old_Graph: %v",g)
	for _,item := range queue_sorted {
		ini := getEndIndex(conections,indexes[item.From]);
        fin := getEndIndex(conections,indexes[item.To]);
        if ini != fin {
            new_Graph.Add(item)
            if((-conections[ini])>(-conections[fin])) {
                conections[ini]+=conections[fin];
                conections[fin]=ini;
            }else{
                conections[fin]+=conections[ini];
                conections[ini]=fin;
            }
        }else{
            log.Printf("ya estan en el mismo arbol - forma ciclo: %v\n",item)
        }
	}
	log.Printf("new_Graph: %v",new_Graph)
	return new_Graph
}

func getEndIndex(arr []int, i int) int {
	index := i
    for (arr[index]>=0) {
        index = arr[index]
    }
	return index;
}

func GraphToJson(g Graph) string {
	json_format,_ := json.Marshal(g)
	return string(json_format)
}

func JsonToGraph(json_format []byte) *Graph {
	var decoded *Graph
	json.Unmarshal(json_format,&decoded)
	return decoded
}

File: Python/kruskal.py
import json
from operator import itemgetter

def GetIndexes(graph):
	index = 0
	indexedMap = {}
	print("edges: ",graph["edges"])
	edges = graph["edges"]
	for i in range (0,len(edges)):
		print("edge: ",edges[i])
		edge = edges[i]
		indexedMap[edge["from"]["value"]] = 0
		print("indexFrom: ",edge["from"]["value"])
		print("indexFrom map: ",indexedMap.get(edge["from"]["value"]))
		indexedMap[edge["to"]["value"]] = 0
		print("indexTo: ",edge["to"]["value"])
		print("indexTo map: ",indexedMap.get(edge["to"]["value"]))
	
	for key in indexedMap:
		print("key: ",key, " value: ",indexedMap[key])
		indexedMap[key] = index
		index = index + 1
		
	return indexedMap

def getEndIndex(arr, i):
	index = i
	while arr[index] >= 0:
		index = arr[index]

	return index

def ApplyKruskal(graph):
	new_Graph = []
	indexes = GetIndexes(graph)
	size = len(indexes)
	conections = []
	for i in range(0,size):
		conections.append(-1)

	print("-- ApplyKruskal --")
	print("indexes: ", indexes)
	print("conections: ", conections)
	
	queue_sorted = sorted(graph["edges"],key=itemgetter('weight'))
	print(indexes)
	print("old_Graph: ",graph)
	print "\n"
	print("sorted: ",queue_sorted)
	for i in range (0,len(queue_sorted)):
		item = queue_sorted[i]
		print item
		ini = getEndIndex(conections,indexes.get(item["from"]["value"]));
		fin = getEndIndex(conections,indexes.get(item["to"]["value"]));
		if (ini != fin):
			new_Graph.append(item)
			if((-conections[ini])>(-conections[fin])):
				conections[ini]+=conections[fin]
				conections[fin]=ini
			else:
				conections[fin]+=conections[ini]
				conections[ini]=fin
		else:
			print("ya estan en el mismo arbol - forma ciclo: ",item)

	print("new_Graph: ",new_Graph)
	return new_Graph